x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -1581,6 +1592,16 @@ public List<AggregationSpec> getAggregations() {
         );
     }
 
+    @Override
+    public Map<String, Mapper.TypeParser> getMappers() {
+        return Map.of(SparseScoredTermsMapper.CONTENT_TYPE, new FieldMapper.TypeParser((n, c) -> {
+            if (SPARSE_SCORED_TERMS_FIELD_MAPPER.check(getLicenseState()) == false) {
+                throw LicenseUtils.newComplianceException(SPARSE_SCORED_TERMS_FIELD_MAPPER.getName());
```

Comment:
If the license expires after this new mapping type has been used on indices in the cluster, what happens to those indices? Will the cluster start up? Can other fields of those indices be searched?

Is there a precedent for platinum mapping types anywhere else in Elasticsearch?

Reply:
These are excellent points @droberts195 and I don't think plat licensing should be on the mapped type. I am curious as to what @jpountz && @javanna think.

Reply:
We can discuss options if we really feel like the field to be platinum, but keeping it in basic would definitely make things simpler wrt the scenario than Dave described.